app/models/user.rb

```diff
@@ -75,4 +75,19 @@ def requires_no_invitation_code!
   def requires_no_invitation_code?
     !!@requires_no_invitation_code
   end
+
+  def undfined_agent_types
+    agents.unscoped.group(:type).pluck(:type).select do |type|
```

Comment:
I thought it would only remove the default scopes but it also removes the User scoping. `reorder('')` does to job we need (remove the default ordering).